### PR TITLE
adding 'OptionsGet' func

### PR DESCRIPTION
**Pull request by Shane-XB-Qian**
let user be easier to access lsp option, specially in func.

### CODE REVIEW

File: plugin/lsp.vim [hunk 1]
```diff
@@ -14,6 +14,10 @@ def g:LspOptionsSet(opts: dict<any>)
   options.OptionsSet(opts)
 enddef
 
+def g:LspOptionsGet(key: string): any
```
Comment (yegappan) [inline plugin/lsp.vim:17]: What is use case for this function?  Can you update doc/lsp.txt with a description of this function and how to use it?

Comment (Shane-XB-Qian) [reply plugin/lsp.vim:17]: the usage mostly is for coding, since it cannot directly access lsp#options#lspOptions in user func.

**Comment by Shane-XB-Qian** [comment]
any other concerns to merge this?

**Comment by yegappan** [comment]
The LspOptionsSet() function accepts a Dict of options and sets them.  Similarly the LspOptionsGet() function should return a Dict of options.   So instead of returning the value of a particular option, can you change this function to return the entire lspOptions Dict? The user can extract the desired key/value from this Dict.

**Comment by Shane-XB-Qian** [comment]
update: now changed to return entire dict as you wished.

**Pull request merged**